parlai/agents/fid/fid.py

```diff
@@ -348,7 +348,13 @@ def __init__(self, opt: Opt, shared: TShared = None):
                 'GoldDocRetrieverFiDAgent only works with `rag_retriever_query` being `"full_history"`. '
                 f'Changing opt value for `rag_retriever_query`: `"{prev_sel}"` -> `"full_history"`'
             )
-
+        if (
+            opt['dynamic_batching'] == 'full'
+            or opt.get('eval_dynamic_batching') == 'full'
+        ):
+            raise RuntimeError(
+                "For now dynamic batching doesn't work with ObservationEchoRetriever as it cleans up _saved_docs mapping after each batch act."
+            )
```

Comment:
note that `dynamic_batching` can also be `'batchsort'`

Reply:
Thanks for pointing that out. fixed